sound/soc/intel/common/soc-acpi-intel-hda-match.c

```diff
@@ -30,6 +30,11 @@ struct snd_soc_acpi_mach snd_soc_acpi_intel_hda_machines[] = {
 		 */
 		.pdata = &hda_pdata,
 	},
+	[SND_SOC_ACPI_INTEL_HDA_MACH_SOF] = {
+		/* SOF HDA machine driver */
+		.drv_name = "skl_hda_dsp_generic",
+		.sof_tplg_filename = "sof-hda-generic.tplg",
+	},
```

Comment:
I am still not very hot on this, I am not sure if this is required, let's ping @crojewsk @crojewsk-intel again on whether SKL is still in use with HDaudio.

Reply:
Best to my knowledge, starting with v5.15 there is not a single system utilizing the skylake-driver regardless of the configuration we would be speaking of. I probably should have sorted out the dsp-driver selection on upstream earlier as we're in v6.10 already. Will prepare something either this or next week.

ACK for ignoring the skylake-driver here.

Reply:
sounds good thanks @crojewsk-intel 

@brentlu we can simplify this assuming only SOF uses this machine driver.

Reply:
I've uploaded new patch which removes HDA external codec support from Skylake driver.